pkg/networkservice/common/cidr/cidr_test.go

```diff
@@ -0,0 +1,158 @@
+		metadata.NewClient(),
+		cidr.NewClient(prefixLen, family),
+		/* Server part */
+		adapters.NewServerToClient(
+			server,
+		),
+	)
+}
+
+func validateConn(t *testing.T, conn *networkservice.Connection, prefix, route string) {
+	require.Equal(t, conn.Context.IpContext.ExtraPrefixes[0], prefix)
+	require.Equal(t, conn.Context.IpContext.SrcRoutes, []*networkservice.Route{
+		{
+			Prefix: route,
+		},
+	})
+}
+
+//nolint:dupl
+func TestIPv4(t *testing.T) {
+	prefixes := []string{"192.168.0.0/16"}
+	server := newServer(prefixes, nil)
+
+	request := newRequest()
+	client1 := newClient(24, networkservice.IpFamily_IPV4, server)
+	conn1, err := client1.Request(context.Background(), request)
+	require.NoError(t, err)
+	validateConn(t, conn1, "192.168.0.0/24", "192.168.0.0/16")
+
+	// refresh
+	conn1, err = client1.Request(context.Background(), request)
+	require.NoError(t, err)
+	validateConn(t, conn1, "192.168.0.0/24", "192.168.0.0/16")
+
+	client2 := newClient(24, networkservice.IpFamily_IPV4, server)
+	conn2, err := client2.Request(context.Background(), newRequest())
+	require.NoError(t, err)
+	validateConn(t, conn2, "192.168.1.0/24", "192.168.0.0/16")
+
+	_, err = client1.Close(context.Background(), conn1)
+	require.NoError(t, err)
+
+	client3 := newClient(24, networkservice.IpFamily_IPV4, server)
+	conn3, err := client3.Request(context.Background(), newRequest())
+	require.NoError(t, err)
+	validateConn(t, conn3, "192.168.0.0/24", "192.168.0.0/16")
+
+	client4 := newClient(24, networkservice.IpFamily_IPV4, server)
+	conn4, err := client4.Request(context.Background(), newRequest())
+	require.NoError(t, err)
+	validateConn(t, conn4, "192.168.2.0/24", "192.168.0.0/16")
+}
+
+//nolint:dupl
+func TestIPv6(t *testing.T) {
+	prefixes := []string{"2001:db8::/96"}
+	server := newServer(prefixes, nil)
+
+	request := newRequest()
+	client1 := newClient(112, networkservice.IpFamily_IPV6, server)
+	conn1, err := client1.Request(context.Background(), request)
+	require.NoError(t, err)
+	validateConn(t, conn1, "2001:db8::/112", "2001:db8::/96")
+
+	// refresh
+	conn1, err = client1.Request(context.Background(), request)
+	require.NoError(t, err)
+	validateConn(t, conn1, "2001:db8::/112", "2001:db8::/96")
+
+	client2 := newClient(112, networkservice.IpFamily_IPV6, server)
+	conn2, err := client2.Request(context.Background(), newRequest())
+	require.NoError(t, err)
+	validateConn(t, conn2, "2001:db8::1:0/112", "2001:db8::/96")
+
+	_, err = client1.Close(context.Background(), conn1)
+	require.NoError(t, err)
+
+	client3 := newClient(112, networkservice.IpFamily_IPV6, server)
+	conn3, err := client3.Request(context.Background(), newRequest())
+	require.NoError(t, err)
+	validateConn(t, conn3, "2001:db8::/112", "2001:db8::/96")
+
+	client4 := newClient(112, networkservice.IpFamily_IPV6, server)
+	conn4, err := client4.Request(context.Background(), newRequest())
+	require.NoError(t, err)
+	validateConn(t, conn4, "2001:db8::2:0/112", "2001:db8::/96")
+}
```

Comment:
This looks as code dup.

Please consider to use test-cases go pattern.

Moreover try to use newClient, validateConn, newServer inline.